projects/overlapping-long-subsessions.kp/transform.py

```diff
@@ -0,0 +1,46 @@
+from collections import namedtuple
+from utils import seconds_to_days, days_diff_from_strs
+from math import floor
+
+def long_to_wide(sqlContext, df, columns):
+    df.createOrReplaceTempView("temp")
+    return sqlContext.sql(
+    """
+         SELECT {0},
+         FIRST(normalized_channel) as normalized_channel,
+         COLLECT_LIST(subsession_start_date) as subsession_start_date,
+         COLLECT_LIST(subsession_length) as subsession_length
+         FROM
+         (
+           SELECT *
+           FROM temp
+           ORDER BY subsession_start_date
```

Comment:
I don't think this is the best way to sort this data; what about sorting it by `subsessionStartDate` and `profileSubsessionCounter`? 

Let's say there are two submissions for a certain client:
- submission *x*, for a sub-session that started at 2 pm on day *d* and ended after 1 hour with `profileSubsessionCounter` = 6.
- submission *y*, for a sub-session that started at 6 pm on day *d* and ended after 25 hours with `profileSubsessionCounter` = 7.

As `subsession_start_date` has a daily precision both submissions are going to have the same start date. Since you are sorting just by `submission_start_date`, you might as well get submission *y* before *x* in your wide view. If that happens, the two submissions will be marked as overlapping even though they aren’t.